pkg/sdk/internal/api/initializer.go

```diff
@@ -2,45 +2,108 @@ package api
 
 import (
 	"fmt"
+	"github.com/benbjohnson/clock"
 	keptnapi "github.com/keptn/go-utils/pkg/api/utils"
+	"github.com/keptn/go-utils/pkg/sdk/connector/controlplane"
+	"github.com/keptn/go-utils/pkg/sdk/connector/eventsource"
+	eventsourceHttp "github.com/keptn/go-utils/pkg/sdk/connector/eventsource/http"
+	eventsourceNats "github.com/keptn/go-utils/pkg/sdk/connector/eventsource/nats"
+	"github.com/keptn/go-utils/pkg/sdk/connector/logforwarder"
+	"github.com/keptn/go-utils/pkg/sdk/connector/logger"
+	"github.com/keptn/go-utils/pkg/sdk/connector/nats"
+	"github.com/keptn/go-utils/pkg/sdk/connector/subscriptionsource"
 	"github.com/keptn/go-utils/pkg/sdk/internal/config"
 	"net/http"
 	"net/url"
 	"strings"
 )
 
-// Initializer implements both methods of creating a new keptn API with internal or remote execution plane
-type Initializer struct {
-	Remote   func(baseURL string, options ...func(*keptnapi.APISet)) (*keptnapi.APISet, error)
-	Internal func(client *http.Client, apiMappings ...keptnapi.InClusterAPIMappings) (*keptnapi.InternalAPISet, error)
+type InitializationResult struct {
+	KeptnAPI            keptnapi.KeptnInterface
+	ControlPlane        *controlplane.ControlPlane
+	EventSenderCallback controlplane.EventSender
+	ResourceHandler     *keptnapi.ResourceHandler
+	Error               error
 }
 
-func CreateKeptnAPI(httpClient *http.Client, env config.EnvConfig) (keptnapi.KeptnInterface, error) {
-	return createAPI(httpClient, env, Initializer{keptnapi.New, keptnapi.NewInternal})
-}
-
-func createAPI(httpClient *http.Client, env config.EnvConfig, apiInit Initializer) (keptnapi.KeptnInterface, error) {
+// Initialize takes care of creating the API clients and initializing the cp-connector library based
+// on environment variables
+func Initialize(env config.EnvConfig, clientFactory HTTPClientGetter, logger logger.Logger) *InitializationResult {
```

Comment:
since a possible error returned by this function is embedded into the `InitializationResult`, should we return a `InitializationResult` value instead of a pointer instead? Otherwise the caller would always need to check for a possible nil pointer returned by this function

Reply:
Good point. Are you okay if i just change it to return `(*InitializationResult,error)`  ?